File: CS/MapArbitraryDatabaseExample/Form1.cs
using System;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;
using System.Windows.Forms;

namespace MapArbitraryDatabaseExample {
    public partial class Form1 :Form {
        public Form1() {
            InitializeComponent();
            BindCategories();
            BindProducts();
        }

        void BindCategories() {
            gridControl1.DataSource = new XPCollection(session1, session1.GetClassInfo(string.Empty, "Category"));
        }

        void BindProducts() {
            SimpleDataLayer dataLayer = (SimpleDataLayer)session1.DataLayer;
            ConnectionProviderSql provider = (ConnectionProviderSql)dataLayer.ConnectionProvider;
            DBTable[] tables = provider.GetStorageTables("Product");
            gridControl2.DataSource = new XPCollection(session1, Program.AddClass(session1.Dictionary, tables[0]));
        }
    }
}

File: VB/MapArbitraryDatabaseExample/Form1.vb
Imports System
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB
Imports System.Windows.Forms

Namespace MapArbitraryDatabaseExample
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            BindCategories()
            BindProducts()
        End Sub

        Private Sub BindCategories()
            gridControl1.DataSource = New XPCollection(session1, session1.GetClassInfo(String.Empty, "Category"))
        End Sub

        Private Sub BindProducts()
            Dim dataLayer As SimpleDataLayer = CType(session1.DataLayer, SimpleDataLayer)
            Dim provider As ConnectionProviderSql = CType(dataLayer.ConnectionProvider, ConnectionProviderSql)
            Dim tables() As DBTable = provider.GetStorageTables("Product")
            gridControl2.DataSource = New XPCollection(session1, Program.AddClass(session1.Dictionary, tables(0)))
        End Sub
    End Class
End Namespace